IANotificationsTestProject/In App Notifications/NoticeView.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+
+class NoticeView: UIView {
+    
+    // MARK: Properties
+
+    @IBOutlet weak var stackView: UIStackView!
+    @IBOutlet weak var noticeLabel: UILabel!
+    @IBOutlet weak var noticeButton: UIButton!
+
+    var delegate: NoticePresentingDelegate?
+    var action: (() -> Void)? {
+        didSet {
+            noticeButton.isHidden = action == nil ? true : false
+        }
+    }
+
+    // MARK: Initialization
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupView()
+    }
+    
+    // MARK: View Layout
+    
+    private func setupView() {
+        let nib = UINib(nibName: "SnackbarNoticeView", bundle: nil)
+        if let view = nib.instantiate(withOwner: self, options: nil).first as? UIView {
```

Comment:
You can probably invert this (`guard let else fatalError()`).

The NIB not loading could be considered as fatal (and you'd want the app to crash immediately).

This, of course, is in a gray area. Some folks may think otherwise!